### ISSUE TITLE
Modifying code to reproduce the Gaussian Adapter paper

### ISSUE THREAD

**Issue by AjayTalati** [jfutoma/MGP-RNN#2]
Hi Joe,

I'm working on simplifying your code here to reproduce the original [GP-Adapter paper](https://arxiv.org/pdf/1606.04443.pdf), with the public [UWaveGestureLibrary dataset](http://www.cs.ucr.edu/~eamonn/time_series_data/) they use for their experiments. 

Basically I've just taken out the baseline co-variates, and medications things, to get a simple skeleton working.

I'm kind of stuck thought? On line [168](https://github.com/jfutoma/MGP-RNN/blob/master/mgp-rnn-fit.py#L168), `get_GP_samples` calls `draw_GP`, with 

`GP_draws = draw_GP(Yi,Ti,Xi,ind_kfi,ind_kti)`

but the signature on line [74](https://github.com/jfutoma/MGP-RNN/blob/master/mgp-rnn-fit.py#L74) is,

`def draw_GP(length,noises,Lf,Kf,Yi,Ti,Xi,ind_kfi,ind_kti):`

So I just wonder how do the trainable GP parameters, 

``` 
    log_length = tf.Variable(tf.random_normal([1],mean=1,stddev=0.1),name="GP-log-length") 
    length = tf.exp(log_length)
    
    #different noise level of each task
    log_noises = tf.Variable(tf.random_normal([M],mean=-2,stddev=0.1),name="GP-log-noises")
    noises = tf.exp(log_noises)
 
   #init cov between tasks. we'll optimize some low rank approx    
    L_f_init = tf.Variable(tf.eye(M),name="GP-Lf")
    Lf = tf.matrix_band_part(L_f_init,-1,0)
    Kf = tf.matmul(Lf,tf.transpose(Lf))
```

get there? 

I can send you my code if you want something that nearly runs? It's got the `UWaveGestureLibrary` dataset pre-processed, into `labels, times, values`?

Thanks very much for your help :+1: 

PS - Caveat, I'm a bit of a Tensorflow beginner, I've been using PyTorch mainly.

**Comment by AjayTalati** [comment]
OK - that was dumb question? Replacing,

`GP_draws = draw_GP(Yi,Ti,Xi,ind_kfi,ind_kti)`

with 

`GP_draws = draw_GP( length, noises, Lf, Kf, Yi, Ti, Xi, ind_kfi, ind_kti )`

on line 168, and also adding `n_mc_smps = 10`, seems to get things working

**Comment by AjayTalati** [comment]
Hi Joe,

closing this as I think it's going to be easier to just setup the simplest example possible using a GP "simulator". Like you said it doesn't have to be realistic it just needs to show how the code works. 

Building on that I should be able to work on porting things over to PyTorch. Life's too short to debug TF :-1:
